### PR TITLE
#157731622: Export User workout in JSON format

### PR DESCRIPTION
**Pull request by King-Benx**
### What does this PR do?
Enable export of a user's workout schedule in JSON format
#### Description of Task to be completed?
Previously, wger only supports transferring a user's workout schedule in PDF format which cannot be used by other applications.

This PR allows a user's workout information to be transferable by having their data exported in JSON format that supports interoperability.

#### Screenshots
screen shot 2018-06-17 at 13 23 40
The imported workouts link in navigation

admin_workouts
The user who wants to export file

export
The modal window for export of json files

imported_workouts
The view for imported json files

screen shot 2018-06-16 at 04 19 51
The exported file

workouts
A workout that has been adopted by a user who imported a json workout

#### How should this be manually tested?
1. Follow the readme.rst procedures on how to set up the application locally 
    NB: Make sure you run the migrations command
2. Login into the system
3. From the navigation bar select workouts or Imported workouts to go to the imports view directly.
4. Select your workout.
5. On the right-hand side, open the options menu and select export/import  JSON
6. Select a tab depending on desired action. If export the file will be downloaded to your /tmp/ folder where you can move it to another location
7. On importing and exporting, you will automatically be redirected to the import and export view

#157731622

### CODE REVIEW

**Review by emugaya — DISMISSED** [review]
I don't see an implementation of import workouts. Implement the whole story.

**Comment by emugaya** [comment]
Make the export Button explicit on what it does. The current button doesn't specify what is being exported.

**Comment by hariclerry** [comment]
Great work so far, you should maybe add the missing test because coveralls is still complaining.

**Comment by emugaya** [comment]
@King-Benx Lets have the imported workouts be shown in on the same screen that the ones a user created are. You need to have a filter for that. Secondly, do make sure that a user is able to see the name of the person they imported the workout from.

File: wger/manager/views/workout.py [hunk 1]
```diff
@@ -121,6 +130,165 @@ def view(request, pk):
...
+#                                     workout_days.day.add(day_of_week)
+#                     if key == 'set_info':
+#                         for info in value:
+#                             for v in info.values():
+#                                 for item in v:
+#                                     for a,b in item.items(): 
+                                        
+#                                         print('##############################')
+#                                         print(a,':::::::::::::::',b)
+#                                         print('##############################')
```
Comment (hariclerry) [inline wger/manager/views/workout.py]: Please remove the print statements from these lines.

**Review by hariclerry — DISMISSED** [review]

File: wger/manager/views/workout.py (hunk 1 above)
Comment (King-Benx) [reply wger/manager/views/workout.py]: Thanks, I was debugging but it was resolved.

File: wger/manager/urls.py [hunk 2]
```diff
@@ -85,6 +85,9 @@
     url(r'^(?P<pk>\d+)/ical$',
         ical.export,
         name='ical'),
+    url(r'^(?P<pk>\d+)/export_json/$', workout.export_json, name='export_json'),
+    url(r'^view_imports$', workout.view_imports, name='view_imports'),
+    # url(r'^import_workout/(?P<imported_from>[0-9A-Za-z_\-]+)/$', workout.import_workout, name='import_workout'),
```
Comment (peterwade153) [inline wger/manager/urls.py]: commented out code?

File: wger/manager/views/workout.py [hunk 3]
```diff
@@ -123,6 +132,165 @@ def view(request, pk):
...
+                                                table_content += '<p><b>Exercise: </b>' + \
+                                                    str(information) + '</p>'
+                                            if key == 'settings':
+                                                table_content += '<p><b>Reps: </b>' + \
+                                                    '-'.join(information) + '</p>'
+                                            if key == 'comments':
+                                                table_content += '<p><b>Comments: </b>' + \
+                                                    '<br>'.join(information) + '</p><hr>'
+                            table_content += '</td>'
+                            # table_content += '<td><a href="' + reverse('manager:workout:import_workout', kwargs={
```
Comment (peterwade153) [inline wger/manager/views/workout.py]: commented out code?

**Pull request merged**